Explorer/Assets/DCL/InWorldCamera/InWorldCamera/Systems/CaptureScreenshotSystem.cs

```diff
@@ -64,30 +76,46 @@ protected override void Update(float t)
 
             if (recorder.State == RecordingState.SCREENSHOT_READY && metadataBuilder.MetadataIsReady)
             {
-                // TODO (Vit): This is a temporary solution for debug purposes. Will be replaced by proper MVC + sending to backend
-                {
-                    hud.Screenshot = recorder.GetScreenshotAndReset();
-                    hud.Metadata = metadataBuilder.GetMetadataAndReset();
-
-                    cameraReelStorageService.UploadScreenshotAsync(hud.Screenshot, hud.Metadata, ctx.Token).Forget();
-
-                    hud.Canvas.enabled = true;
-                }
-
+                ProcessCapturedScreenshot();
                 return;
             }
 
-            if (recorder.State == RecordingState.IDLE && World.TryGet<InWorldCameraInput>(camera, out var input) && input.TakeScreenshot)
+            if (ScreenshotIsRequested())
             {
-                hud.Canvas.enabled = false;  // TODO (Vit): This is a temporary solution for debug puproses. Will be replaced by proper MVC
+                hudController.Hide(isInstant: true);
                 coroutineRunner.StartCoroutine(recorder.CaptureScreenshot());
                 CollectMetadata();
             }
         }
 
+        private void ProcessCapturedScreenshot()
+        {
+            screenshot = recorder.GetScreenshotAndReset();
+            metadata = metadataBuilder.GetMetadataAndReset();
+
+            cameraReelStorageService.UploadScreenshotAsync(screenshot, metadata, ctx.Token).Forget();
+
+            hudController.Show();
+            hudController.PlayScreenshotFX(screenshot, SPLASH_FX_DURATION, MIDDLE_PAUSE_FX_DURATION, IMAGE_TRANSITION_FX_DURATION);
+            hudController.DebugCapture(screenshot, metadata);
+        }
+
+        private bool ScreenshotIsRequested()
+        {
+            if (recorder.State != RecordingState.IDLE) return false;
+
+            if (World.Has<TakeScreenshotUIRequest>(camera))
+            {
+                World.Remove<TakeScreenshotUIRequest>(camera);
+                return true;
+            }
+
+            return World.TryGet(camera, out InWorldCameraInput input) && input.TakeScreenshot;
```

Comment:
Question: you check both the InWorldCameraInput and the TakeScreenshotUIRequest, isn't there a way to unify both? Like creating a TakeScreenshotRequest from both places and checking only for that struct here?